### PR TITLE
Add powershell support

### PR DESCRIPTION
**Pull request by Mike325**

### CODE REVIEW

**Comment by mhinz** [comment]
I can't reproduce that on Win 10. What are your shell settings within Nvim? Just putting `set shell=powershell` into my init.vim wasn't enough.

**Comment by Mike325** [comment]
This should works with neovim, I'm also using  win10 with neovim 0.3.4
```viml
set shell=powershell.exe
set shellcmdflag=-NoLogo\ -NoProfile\ -ExecutionPolicy\ RemoteSigned\ -Command
set shellxquote=
let &shellquote = '"'
let &shellpipe  = '| Out-File -Encoding UTF8 %s'
let &shellredir = '| Out-File -Encoding UTF8 %s'
```

**Comment by mhinz** [comment]
Okay, I can reproduce it with these settings!

File: autoload/sy/repo.vim [hunk 1]
```diff
@@ -297,7 +297,7 @@ function! s:initialize_job(vcs) abort
   let vcs_cmd = s:expand_cmd(a:vcs, g:signify_vcs_cmds)
   if has('win32')
     if has('nvim')
-      let cmd = &shell =~ 'cmd' ? vcs_cmd : ['sh', '-c', vcs_cmd]
+      let cmd = &shell =~ '\(cmd\|powershell\)\(\.exe\)\?' ? vcs_cmd : ['sh', '-c', vcs_cmd]
```
Comment (mhinz) [inline autoload/sy/repo.vim]:
```suggestion
      let cmd = &shell =~ '\v%(cmd|powershell)' ? vcs_cmd : ['sh', '-c', vcs_cmd]
```

Comment (Mike325) [reply autoload/sy/repo.vim]: Done !

**Comment by mhinz** [comment]
Thank you! ✨

**Pull request merged**